src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -417,24 +419,46 @@ public async Task<JObject> Resolve(InvocationExpressionSyntax method, Dictionary
+                            // if more, it will ensure to produce an error
+                            var optionalParamsCnt = methodParamsInfo.Count(p => p.HasDefaultValue);
+                            if (passedParamCnt <= methodParamsInfo.Count &&
+                                passedParamCnt + optionalParamsCnt >= methodParamsInfo.Count)
+                                passedParamCnt = methodParamsInfo.Count;
+                        }
+                        commandParamsObjWriter.Write(passedParamCnt + isTryingLinq);
                         if (isTryingLinq == 1)
                             commandParamsObjWriter.WriteObj(objectId, context.SdbAgent);
-                        foreach (var arg in method.ArgumentList.Arguments)
+                        for (var i = 0; i < passedParamCnt; i++)
                         {
-                            if (arg.Expression is LiteralExpressionSyntax)
+                            // explicitly passed arguments
+                            if (i < method.ArgumentList.Arguments.Count)
                             {
-                                if (!await commandParamsObjWriter.WriteConst(arg.Expression as LiteralExpressionSyntax, context.SdbAgent, token))
-                                    return null;
-                            }
-                            if (arg.Expression is IdentifierNameSyntax)
-                            {
-                                var argParm = arg.Expression as IdentifierNameSyntax;
-                                if (!await commandParamsObjWriter.WriteJsonValue(memberAccessValues[argParm.Identifier.Text], context.SdbAgent, token))
-                                    return null;
+                                var arg = method.ArgumentList.Arguments[i];
+                                if (arg.Expression is LiteralExpressionSyntax)
+                                {
+                                    if (!await commandParamsObjWriter.WriteConst(arg.Expression as LiteralExpressionSyntax, context.SdbAgent, token))
+                                        return null;
+                                }
+                                if (arg.Expression is IdentifierNameSyntax)
+                                {
+                                    var argParm = arg.Expression as IdentifierNameSyntax;
+                                    if (!await commandParamsObjWriter.WriteJsonValue(memberAccessValues[argParm.Identifier.Text], context.SdbAgent, token))
+                                        return null;
+                                }
                             }
+                            // optional arguments that were not overritten
+                            else if (!await commandParamsObjWriter.WriteConst(methodParamsInfo[i].TypeCode, methodParamsInfo[i].Value, context.SdbAgent, token))
+                                return null;
```

Comment:
just to make it slightly easier to read.

```suggestion
                            else
                            {
                                if (!await commandParamsObjWriter.WriteConst(methodParamsInfo[i].TypeCode, methodParamsInfo[i].Value, context.SdbAgent, token))
                                    return null;
                            }
```